frontend/lib/src/components/widgets/AudioInput/NoMicPermissions.tsx

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { ReactElement } from "react"
+
+import { ENABLE_PERIPHERALS_DOCS_URL } from "@streamlit/lib/src/urls"
+
+import {
+  StyledNoMicInputContainerDiv,
+  StyledNoMicInputLearnMoreLink,
+  StyledNoMicPermissionsErrorTextSpan,
+} from "./styled-components"
+
+const NoMicPermissions = (): ReactElement => {
+  return (
+    <StyledNoMicInputContainerDiv data-testid="StyledNoMicInputContainerDiv">
```

Comment:
**suggestion:** Similar to the above, this `testid` shouldn't be necessary. Instead the tests should get elements via accessible manners, such as `getByText`